File: Lab №4/Python/Equilateral.py
from TTriangle import *


class Equilateral(TTriangle):

    def __init__(self, point1, point2):
        super().__init__(point1, point2, point1.distance_to(point2), 60)
        self.side3 = round(point1.distance_to(point2), 2)
        self.side1 = self.side2 = self.side3
        self.angle = 60

    def calculate_area(self):
        return round((self.side3 ** 2 * math.sqrt(3) / 4), 2)

    def info(self):
        output = ''
        output += '\nДанi про рiвностороннiй трикутник:\n'
        i = 1
        for point in self.coordinates_list:
            output += 'x' + str(i) + ' = ' + str(point.x) + '; y' + str(i) + ' = ' + str(point.y) + '\n'
            i += 1
        output += 'side1 = side2 = side3 = ' + str(self.side3) + '\nangle = ' + str(self.angle) + '\nПлоща трикутника = '
        output += str(Equilateral.calculate_area(self)) + '\n'
        return output

File: Lab №4/C++/Equilateral.cpp
#include <sstream>
#include "Equilateral.h"

Equilateral::Equilateral(Coordinates point1, Coordinates point2):
	TTriangle(point1, point2, point1.distanceTo(point2), 60)
{
	side3 = round(point1.distanceTo(point2) * 100) / 100;
	side1 = side2 = side3;
	angle = 60;
}

double Equilateral::CalculatePerimeter()
{
	return 0.0;
}

double Equilateral::CalculateArea() {
	return round((pow(side3, 2) * sqrt(3)) / 4 * 100) / 100;
}

string Equilateral::info()   //виведення данних про трикутник
{
	stringstream output;
	output << endl << "Данi про рiвностороннiй трикутник:" << endl;
	int i = 1;
	for (auto point:coordinates)
	{
		output << "x" << i << " = " << point.x << "; y" << i << " = " << point.y << endl;
		i++;
	}
	output << "side1 = side2 = side3 = " << side3 << endl << "angle = " << angle << endl << "Площа трикутника = " << CalculateArea() << endl;
	return output.str();
}
